rollup.config.js

```diff
@@ -0,0 +1,173 @@
+import fs from 'fs';
+import path from 'path';
+import babel from '@rollup/plugin-babel';
+import resolve from '@rollup/plugin-node-resolve';
+import svgr from '@svgr/rollup';
+import { terser } from 'rollup-plugin-terser';
+import urlPlugin from '@rollup/plugin-url';
+
+const extensions = ['.ts', '.tsx'];
+
+const { name } = require(path.resolve(process.cwd(), 'package.json'));
+
+function getAllPackages(dir) {
+  const dirList = fs.readdirSync(dir);
+
+  return dirList
+    .map(subDir => `${path.resolve(dir, subDir)}/package.json`)
+    .filter(packageJsonPath => fs.existsSync(packageJsonPath))
+    .map(packageJsonPath => require(packageJsonPath).name);
+}
+
+function getLodashExternals() {
+  return fs
+    .readdirSync(path.resolve(__dirname, 'node_modules', 'lodash'))
+    .filter(fileName => {
+      const splitName = fileName.split('.');
+      const extension = splitName[splitName.length - 1];
+
+      return extension === 'js';
+    })
+    .map(fileName => 'lodash/' + fileName.split('.')[0]);
+}
+
+function getDirectGlyphImports() {
+  const glyphsDir = path.resolve(__dirname, './packages/icon/src/glyphs');
+
+  return fs
+    .readdirSync(glyphsDir)
+    .filter(path => /.svg/.test(path))
+    .map(
+      fileName => `@leafygreen-ui/icon/dist/${path.basename(fileName, '.svg')}`,
+    );
+}
+
+function getGeneratedFiles() {
+  const directory = path.resolve(process.cwd(), 'src/generated');
+
+  if (!fs.existsSync(directory)) {
+    return [];
+  }
+
+  return fs
+    .readdirSync(directory)
+    .filter(file => /\.tsx?$/.test(file))
+    .map(file => path.resolve(directory, file));
+}
+
+const allPackages = getAllPackages('../../packages');
+const directGlyphImports = getDirectGlyphImports();
+
+// Mapping of packages to the `window` property they'd be
+// bound to if used in the browser without a module loader.
+// This is defined on a best effort basis since not all
+// modules are compatible with being loaded directly.
+const globals = {
+  clipboard: 'ClipboardJS',
+  'highlightjs-graphql': 'hljsDefineGraphQL',
+  polished: 'polished',
+  react: 'React',
+  'react-dom': 'ReactDOM',
+  'prop-types': 'PropTypes',
+  lodash: '_',
+  /**
+   * External dependencies that must be loaded by a module loader
+   *   - lodash/*
+   *   - highlight.js
+   *   - create-emotion
+   *   - create-emotion-server
+   *   - react-transition-group
+   **/
+};
+
+allPackages.forEach(packageName => {
+  globals[packageName] = packageName;
```

Comment:
Feels like we should namespace this to `global.LG[packageName]`? So we don't pollute the global name space?

Reply:
`globals` is a local variable defined above, not to be confused with `global`.

Reply:
Ah, ty.